scipy/io/matlab/_streams.pyx

```diff
@@ -228,7 +228,7 @@ def _read_into(GenericStream st, size_t n):
 def _read_string(GenericStream st, size_t n):
     # for testing only.  Use st.read instead
     cdef void *d_ptr
-    cdef object obj = st.read_string(n, &d_ptr, True)
+    cdef object obj = st.read_string(n, &d_ptr, True)  # no-cython-lint
```

Comment:
Is the use of `# no-cython-lint` here also because of https://github.com/MarcoGorelli/cython-lint/issues/52?

Reply:
It was because of https://github.com/MarcoGorelli/cython-lint/issues/50, but can be removed with the latest cython-lint.

Reply:
Actually, it's good you are taking a look at this. I wasn't sure whether it is safe, in this specific instance, to not assign to obj (which is what the linter wants).

Reply:
I should have looked at the full code before commenting... in this case, `obj` is, in fact, not explicitly used.  However, the test fails if the result is not assigned to something, probably because of a memory lifetime issue.  Instead of adding `# no-cython-lint`, the widely-used linter convention of prefixing the name with `_` could be used.

I also wonder if there is any point to declaring the object as `cdef object`.  In Cython is there any difference between `cdef object x = ...` and just `x = ...`? (That's not a rhetorical question--I don't know.)  The test passes and the linter does not complain if I use

```
    _obj = st.read_string(n, &d_ptr, True)
```
but if you want these changes to be minimal--just what is needed to make cython-lint happy, then
```
    cdef object _obj = st.read_string(n, &d_ptr, True)
```
also works.